test/javascript/components/accordion.js

```diff
@@ -1,20 +1,142 @@
 describe('Accordion', function() {
-	var plugin;
-	var $html;
+  var plugin;
+  var $html;
+  var template = `<ul class="accordion" data-accordion>
+  <li class="accordion-item is-active" data-accordion-item>
+    <a href="#" class="accordion-title">Accordion 1</a>
+    <div class="accordion-content" data-tab-content >
+      <p>Panel 1. Lorem ipsum dolor</p>
+      <a href="#">Nowhere to Go</a>
+    </div>
+  </li>
+  <li class="accordion-item" data-accordion-item>
+    <a href="#" class="accordion-title">Accordion 2</a>
+    <div class="accordion-content" data-tab-content>
+      <textarea></textarea>
+      <button class="button">I do nothing!</button>
+    </div>
+  </li>
+  <li class="accordion-item" data-accordion-item>
+    <a href="#" class="accordion-title">Accordion 3</a>
+    <div class="accordion-content" data-tab-content>
+      Pick a date!
+      <input type="date"></input>
+    </div>
+  </li>
+</ul>`;
 
-	// afterEach(function() {
-	// 	plugin.destroy();
-	// 	$html.remove();
-	// });
+  afterEach(function() {
+    plugin.destroy();
+    $html.remove();
+  });
 
-	describe('constructor()', function() {
-		// it('', function() {
-		// 	$html = $('').appendTo('body');
-		// 	plugin = new Foundation.Accordion($html, {});
+  describe('constructor()', function() {
+    it('stores the element and plugin options', function() {
+      $html = $(template).appendTo('body');
+      plugin = new Foundation.Accordion($html, {});
 
-		// 	plugin.$element.should.be.an('object');
-		// 	plugin.options.should.be.an('object');
-		// });
-	});
+      plugin.$element.should.be.an('object');
+      plugin.options.should.be.an('object');
+    });
+  });
 
+  describe('up()', function() {
+    it('closes the targeted container if allowAllClosed is true', function() {
+      $html = $(template).appendTo('body');
+      plugin = new Foundation.Accordion($html, {allowAllClosed: true});
+
+      plugin.up($html.find('.accordion-content').eq(0));
+      $html.find('.accordion-content').eq(0).should.have.attr('aria-hidden', 'true');
+      $html.on('up.zf.accordion', function() {
+        $html.find('.accordion-content').eq(0).should.be.hidden;
+        done();
+      });
+    });
+
+    it('toggles attributes of title of the targeted container', function() {
+      $html = $(template).appendTo('body');
+      plugin = new Foundation.Accordion($html, {});
+
+      plugin.up($html.find('.accordion-content').eq(1));
+      $html.find('.accordion-title').eq(1).should.have.attr('aria-expanded', 'false');
+      $html.find('.accordion-title').eq(1).should.have.attr('aria-selected', 'false');
```

Comment:
The usage of `toggle()` is a bit limited as it only checks if the target is active and closes it if so, or opens it otherwise. Basically `toggle()` just delegates to `up()` and `down()`.